pkg/imageprovider/rhcos.go

```diff
@@ -84,8 +93,38 @@ func imageKey(data imageprovider.ImageData) string {
 	)
 }
 
+func getIgnitionOverride(imageMetadata *metav1.ObjectMeta, log logr.Logger) ([]byte, error) {
+	if overrideURI, exist := imageMetadata.Annotations[ignitionOverrideAnnotation]; exist {
+		log.Info("using Ignition override when building the image", "host", imageMetadata.Name, "overrideURI", overrideURI)
+		resp, err := http.Get(overrideURI) //#nosec G107
+		if err != nil {
+			return nil, errors.Wrap(err, "could not download Ignition override")
+		}
+		defer resp.Body.Close()
+
+		override, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return nil, errors.Wrap(err, "could not download Ignition override")
+		}
+
+		return override, nil
+	}
+
+	if infraEnvName, useInfraEnv := imageMetadata.Labels[infraEnvLabel]; useInfraEnv {
+		log.Info("host is using an InfraEnv, waiting for Ignition override", "host", imageMetadata.Name, "infraEnv", infraEnvName)
+		return nil, imageprovider.ImageNotReady{}
```

Comment:
Would be good if we can specify a reason in the Ready condition so that it's at least theoretically visible to users. This may require an upstream change though.

Reply:
I think it will. But I agree, I've spotted it in my testing as well.